packages/pangea-sdk/.sdk-ci.yml

```diff
@@ -80,72 +84,71 @@
         ENV: ${SERVICE_VAULT_ENV}
         TEST: test_vault
   rules:
-    - if: '$CLOUD == "GCP" && $TEST == "test_file_scan"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD == "GCP" && $TEST == "test_file_scan"'
       allow_failure: true
-    - if: '$CLOUD == "GCP" && $TEST != "test_file_scan"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD == "GCP" && $TEST != "test_file_scan"'
       allow_failure: false
-    - if: '$CLOUD != "GCP"'
+    - if: $CI_COMMIT_BRANCH && '$CLOUD != "GCP"'
       allow_failure: false
 
-.python-sdk-publish-base:
-  before_script:
-    - cd packages/pangea-sdk/
-    - pip install poetry
-    - poetry check
-    - poetry install
-  rules:
-    - if: $CI_COMMIT_BRANCH == "release"
-      changes:
-        - packages/pangea-sdk/**/*
-      when: on_success
-
-python-sdk-lint:
-  extends: .python-sdk-base
+sdk-lint-black:
+  extends: .sdk-base
   stage: lint
+  needs: []
   script:
     - poetry run black .
 
-python-sdk-generate-docs:
-  extends: .python-sdk-base
+sdk-lint-mypy:
+  extends: .sdk-base
   stage: lint
-  allow_failure: true
+  needs: []
+  script:
+    - poetry run mypy pangea tests
+
+sdk-build:
+  extends: .sdk-base
+  stage: build
+  script:
+    - poetry build
+  artifacts:
+    paths: ["packages/pangea-sdk/dist"]
+    when: on_success
+
+sdk-docs:
+  extends: .sdk-base
+  stage: build
   script:
     - poetry run python parse_module.py > python_sdk.json
   artifacts:
     expire_in: 1 month
     expose_as: python_sdk
-    paths: ['packages/pangea-sdk/python_sdk.json']
+    paths: ["packages/pangea-sdk/python_sdk.json"]
     when: on_success
 
-python-sdk-type-check:
-  extends: .python-sdk-base
-  stage: lint
-  needs: []
-  script:
-    - poetry run mypy pangea tests
-
-python-sdk-unit-tests:
-  extends: .python-sdk-base
+sdk-unit-tests:
+  extends: .sdk-base
   stage: unit_tests
   script:
     - poetry run python -m unittest tests.unit
 
-python-sdk-integration-tests:
-  extends: .python-sdk-test-base
+sdk-integration-tests:
+  extends: .sdk-test-base
   stage: integration_tests
   script:
     - poetry run python -m unittest tests.integration.${TEST}
 
 # TODO: Add asyncio tests when upgrade python 3.8
-# python-sdk-asyncio-integration-tests:
-#   extends: .python-sdk-test-base
+# sdk-asyncio-integration-tests:
+#   extends: .sdk-test-base
 #   stage: integration_tests
 #   script:
 #     - poetry run python -m unittest tests.integration.asyncio.${TEST}
 
-python-sdk-publish:
-  extends: .python-sdk-publish-base
+sdk-publish:
+  extends: .sdk-base
   stage: publish
+  needs: [sdk-build]
```

Comment:
Don't we want to add `sdk-integration-tests` as `needs` too? Or they won't run because it's a tag push and not a commit?

Reply:
Right, there's no need for integration tests on the tag push because they would've already executed (successfully, we hope) on the actual code changes that precede the tag.